### PR TITLE
ROOT 6-22 - get latest commits

### PR DESCRIPTION
**Pull request by mrodozov**
we are one commit behind

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
I was running this to figure out why this tests are failing:
```

/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmsdist/6841/14611/install.sh
cd CMSSW_12_0_X_2021-04-26-2300 ; cmsenv
# first run runTheMatrix to get the cmsDriver commands
export CMS_PATH=/cvmfs/cms-ib.cern.ch # you need this to make eos files available for ibs visible to xrootd
runTheMatrix -i all -l 136.8311 -t 4
# when it fails check the step2 command
cat 136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017/cmdLog
# run the second step
 cmsDriver.py step2  --conditions auto:run2_data_relval -s PAT,DQM:@miniAODDQM --process PAT -n 100 --data  --era Run2_2017,run2_miniAOD_94XFall17 --eventcontent MINIAOD,DQM --scenario pp --datatier MINIAOD,DQMIO --filein filelist:step1_dasquery.log --lumiToProcess step1_lumiRanges.log --fileout file:step2.root  --nThreads
```
 
 and the output is showing failure in CMS code:

 ```

 2021-04-28 17:34:07.622560: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2 AVX AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-04-28 17:34:07.725788: I tensorflow/core/platform/profile_utils/cpu_utils.cc:112] CPU Frequency: 2194915000 Hz
Begin processing the 1st record. Run 305064, Event 55020723, LumiSection 36 on stream 2 at 28-Apr-2021 17:34:21.987 CEST
Begin processing the 2nd record. Run 305064, Event 54905050, LumiSection 36 on stream 0 at 28-Apr-2021 17:34:21.989 CEST
Begin processing the 3rd record. Run 305064, Event 54823206, LumiSection 36 on stream 3 at 28-Apr-2021 17:34:21.990 CEST
Begin processing the 4th record. Run 305064, Event 54870566, LumiSection 36 on stream 1 at 28-Apr-2021 17:34:21.991 CEST
cmsRun: /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-04-26-2300/src/FWCore/Concurrency/src/WaitingTaskList.cc:57: void edm::WaitingTaskList::reset(): Assertion `m_head == nullptr' failed.
```

@makortel

**Comment by makortel** [comment]
Here is stack trace from the pat1 test
```
27-Apr-2021 21:00:48 CEST  Successfully opened file root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/relval/CMSSW_9_2_2/RelValProdTTbar_13/AODSIM/91X_mcRun2_asymptotic_v3-v1/10000/EEB99F74-DA4D-E711-A41C-0025905A48F2.root
Begin processing the 1st record. Run 1, Event 2, LumiSection 1 on stream 0 at 27-Apr-2021 21:01:27.341 CEST
cmsRun: /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-04-26-2300/src/FWCore/Concurrency/src/WaitingTaskList.cc:57: void edm::WaitingTaskList::reset(): Assertion `m_head == nullptr' failed.

Thread 1 (Thread 0x2b6072894a80 (LWP 14347)):
#0  0x00002b6070c8dccd in poll () from /lib64/libc.so.6
#1  0x00002b6079bcdcd7 in full_read.constprop () from /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmsdist/6841/14611/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/pluginFWCoreServicesPlugins.so
#2  0x00002b6079bce56c in edm::service::InitRootHandlers::stacktraceFromThread() () from /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmsdist/6841/14611/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/pluginFWCoreServicesPlugins.so
#3  0x00002b6079bcf922 in sig_dostack_then_abort () from /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmsdist/6841/14611/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/pluginFWCoreServicesPlugins.so
#4  <signal handler called>
#5  0x00002b6070bd03d7 in raise () from /lib64/libc.so.6
#6  0x00002b6070bd1ac8 in abort () from /lib64/libc.so.6
#7  0x00002b6070bc91a6 in __assert_fail_base () from /lib64/libc.so.6
#8  0x00002b6070bc9252 in __assert_fail () from /lib64/libc.so.6
#9  0x00002b606e159d19 in edm::WaitingTaskList::reset() () from /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmsdist/6841/14611/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreConcurrency.so
#10 0x00002b606e4b13bb in edm::Worker::reset() () from /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmsdist/6841/14611/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreFramework.so
#11 0x00002b606e4afd64 in edm::WorkerManager::resetAll() () from /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmsdist/6841/14611/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreFramework.so
#12 0x00002b606e3b6240 in edm::StreamSchedule::processOneStreamAsync<edm::OccurrenceTraits<edm::LuminosityBlockPrincipal, (edm::BranchActionType)2> >(edm::WaitingTaskHolder, edm::OccurrenceTraits<edm::LuminosityBlockPrincipal, (edm::BranchActionType)2>::TransitionInfoType&, edm::ServiceToken const&, bool)::{lambda()#2}::operator()() () from /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmsdist/6841/14611/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreFramework.so
#13 0x00002b606e3b6574 in tbb::internal::function_task<edm::StreamSchedule::processOneStreamAsync<edm::OccurrenceTraits<edm::LuminosityBlockPrincipal, (edm::BranchActionType)2> >(edm::WaitingTaskHolder, edm::OccurrenceTraits<edm::LuminosityBlockPrincipal, (edm::BranchActionType)2>::TransitionInfoType&, edm::ServiceToken const&, bool)::{lambda()#3}>::execute() () from /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmsdist/6841/14611/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreFramework.so
#14 0x00002b606fb7d26d in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::process_bypass_loop (this=this@entry=0x2b60735eb200, context_guard=..., t=0x2b60735e3340, isolation=isolation@entry=0) at ../../include/tbb/task.h:1003
#15 0x00002b606fb7d756 in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::local_wait_for_all (this=0x2b60735eb200, parent=..., child=<optimized out>) at ../../src/tbb/scheduler.h:650
#16 0x00002b606e39e1f3 in tbb::internal::task_group_base::wait() () from /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmsdist/6841/14611/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreFramework.so
#17 0x00002b606e38e760 in edm::EventProcessor::endUnfinishedLumi() () from /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmsdist/6841/14611/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreFramework.so
#18 0x00002b606e352b51 in edm::EventProcessor::runToCompletion() [clone .cold] () from /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmsdist/6841/14611/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreFramework.so
#19 0x000000000040bae6 in tbb::interface7::internal::delegated_function<main::{lambda()#1}::operator()() const::{lambda()#1} const, void>::operator()() const ()
#20 0x00002b606fb76552 in tbb::interface7::internal::task_arena_base::internal_execute (this=0x7fff13e841c0, d=...) at ../../src/tbb/arena.cpp:1105
#21 0x000000000040ca13 in main::{lambda()#1}::operator()() const ()
#22 0x000000000040b62c in main ()

Current Modules:

Module: none (crashed)
```

I have no clue on the top of my head. @Dr15Jones, any ideas?

What does the added ROOT commit do?

**Comment by mrodozov** [comment]
nothing much, I suspect this was there all along, but I got carried out 
by the last test in the prev PR to update ROOT 6-22, which went like this - 

https://github.com/cms-sw/cmsdist/pull/6810#issuecomment-823771782

the same tests fail, then I update it again before testing, test and only one unit test was failing.
I didn't get suspicious how does just two (unrelated) commits fix the prev tests and merge the change.
Nothing breaks in the IBs eventually so I don't search for a problem, and I suspect if I merge this it wont show on the IBs, but the problem is there. We can try this on DEVEL

Bottom line - it's not due to this one commit. Also it's not one commit - it's five, but I don't think it's root - 
it's from the commit  [AArch64] Materialize FP constant in ....
on this branch 
https://github.com/cms-sw/root/commits/cms/v6-22-00-patches/54d0000
that we put on top

**Comment by Dr15Jones** [comment]
> I have no clue on the top of my head. @Dr15Jones, any ideas?

Maybe there is a 'runaway' exception? If this is repeatable, try running under gdb with `catch throw` and see if it triggers.

**Comment by mrodozov** [comment]
>  If this is repeatable,

I have three out of three failures + the tests.

this is the catch throw result:

```

Thread 1 "cmsRun" hit Catchpoint 1 (exception thrown), 0x00007ffff5a0532e in __cxxabiv1::__cxa_throw (obj=obj@entry=0x7fff3a43b820, 
    tinfo=0x7ffff5b2b270 <typeinfo for std::length_error>, dest=0x7ffff5a1a4f0 <std::length_error::~length_error()>) at ../../../../libstdc++-v3/libsupc++/eh_throw.cc:78
78	../../../../libstdc++-v3/libsupc++/eh_throw.cc: No such file or directory.
(gdb) where
#0  0x00007ffff5a0532e in __cxxabiv1::__cxa_throw (obj=obj@entry=0x7fff3a43b820, tinfo=0x7ffff5b2b270 <typeinfo for std::length_error>, 
    dest=0x7ffff5a1a4f0 <std::length_error::~length_error()>) at ../../../../libstdc++-v3/libsupc++/eh_throw.cc:78
#1  0x00007ffff59fbcd4 in std::__throw_length_error (__s=0x7ffff7f129d1 "vector::_M_fill_insert") at ../../../../../libstdc++-v3/src/c++11/functexcept.cc:78
#2  0x00007ffff7f0405c in std::vector<void const*, std::allocator<void const*> >::_M_check_len(unsigned long, char const*) const ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libDataFormatsCommon.so
#3  0x00007ffff7f04245 in std::vector<void const*, std::allocator<void const*> >::_M_fill_insert(__gnu_cxx::__normal_iterator<void const**, std::vector<void const*, std::allocator<void const*> > >, unsigned long, void const* const&) ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libDataFormatsCommon.so
#4  0x00007fffbfdc6a6d in ROOT::read_recocLcLPFCandidate_2(char*, TVirtualObject*) ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libDataFormatsParticleFlowCandidate.so
#5  0x00007ffff71b2c25 in int TStreamerInfo::ReadBufferArtificial<char**>(TBuffer&, char** const&, TStreamerElement*, int, int) ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/external/slc7_amd64_gcc900/lib/libRIO.so
#6  0x00007ffff727af7c in int TStreamerInfo::ReadBuffer<char**>(TBuffer&, char** const&, TStreamerInfo::TCompInfo* const*, int, int, int, int, int) ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/external/slc7_amd64_gcc900/lib/libRIO.so
#7  0x00007ffff712d4e4 in TStreamerInfoActions::VectorLooper::GenericRead(TBuffer&, void*, void const*, TStreamerInfoActions::TLoopConfiguration const*, TStreamerInfoActions::TConfiguration const*) () from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/external/slc7_amd64_gcc900/lib/libRIO.so
#8  0x00007ffff7027c64 in TBufferFile::ApplySequence(TStreamerInfoActions::TActionSequence const&, void*, void*) ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/external/slc7_amd64_gcc900/lib/libRIO.so
#9  0x00007ffff7730202 in TBranchElement::GetEntry(long long, int) ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/external/slc7_amd64_gcc900/lib/libTree.so
#10 0x00007ffff772fd0d in TBranchElement::GetEntry(long long, int) ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/external/slc7_amd64_gcc900/lib/libTree.so
#11 0x00007fffb56be568 in edm::RootTree::getEntry(TBranch*, long long) const ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/pluginIOPoolInput.so
#12 0x00007fffb5693f23 in edm::RootDelayedReader::getProduct_(edm::BranchID const&, edm::EDProductGetter const*) ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/pluginIOPoolInput.so
#13 0x00007ffff7ba1f04 in edm::DelayedReader::getProduct(edm::BranchID const&, edm::EDProductGetter const*, edm::ModuleCallingContext const*) ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreFramework.so
#14 0x00007ffff7c6c885 in edm::DelayedReaderInputProductResolver::prefetchAsync_(edm::WaitingTaskHolder, edm::Principal const&, bool, edm::ServiceToken const&, edm::SharedResourcesAcquirer*, edm::ModuleCallingContext const*) const::{lambda()#1}::operator()() const ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreFramework.so
#15 0x00007ffff7c6ca61 in edm::SerialTaskQueue::QueuedTask<edm::SerialTaskQueueChain::push<edm::DelayedReaderInputProductResolver::prefetchAsync_(edm::WaitingTaskHolder, edm::Principal const&, bool, edm::ServiceToken const&, edm::SharedResourcesAcquirer*, edm::ModuleCallingContext const*) const::{lambda()#1}&>(tbb::task_group&, edm::DelayedReaderInputProductResolver::prefetchAsync_(edm::WaitingTaskHolder, edm::Principal const&, bool, edm::ServiceToken const&, edm::SharedResourcesAcquirer*, edm::ModuleCallingContext const*) const::{lambda()#1}&)::{lambda()#1}>::execute() ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreFramework.so
#16 0x00007ffff7e8cf49 in tbb::internal::function_task<edm::SerialTaskQueue::spawn(edm::SerialTaskQueue::TaskBase&)::{lambda()#1}>::execute() ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreConcurrency.so
#17 0x00007ffff645d26d in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::process_bypass_loop (this=this@entry=0x7ffff2147900, context_guard=..., 
--Type <RET> for more, q to quit, c to continue without paging--
    7fff4204ee40, isolation=isolation@entry=0) at ../../include/tbb/task.h:1003
#18 0x00007ffff645d756 in tbb::internal::custom_scheduler<tbb::internal::IntelSchedulerTraits>::local_wait_for_all (this=0x7ffff2147900, parent=..., child=<optimized out>)
    at ../../src/tbb/scheduler.h:650
#19 0x00007ffff7bc1058 in edm::EventProcessor::processLumis(std::shared_ptr<void> const&) ()
   from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreFramework.so
#20 0x00007ffff7bca035 in edm::EventProcessor::runToCompletion() () from /build/mrodozov/root/622_install/CMSSW_12_0_X_2021-04-26-2300/lib/slc7_amd64_gcc900/libFWCoreFramework.so
#21 0x000000000040bae6 in tbb::interface7::internal::delegated_function<main::{lambda()#1}::operator()() const::{lambda()#1} const, void>::operator()() const ()
#22 0x00007ffff6456552 in tbb::interface7::internal::task_arena_base::internal_execute (this=0x7fffffff31b0, d=...) at ../../src/tbb/arena.cpp:1105
#23 0x000000000040ca13 in main::{lambda()#1}::operator()() const ()
#24 0x000000000040b62c in main ()
(gdb) 

```

**Comment by mrodozov** [comment]
checking if ROOT master is the same

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
+externals
